src/components/LyricView.tsx

```diff
@@ -59,29 +59,51 @@ const LyricView: React.FC<LyricViewProps> = (props: LyricViewProps) => {
     let key = 0;
 
     const songLyrics = song.lyrics;
-    const verseKeys = Object.keys(songLyrics);
-    const lyricBlocks: string[][] = Object.values(songLyrics);
+    const presentationOrder = song.presentation?.split(" ");
 
-    for (let i = 0; i < verseKeys.length; ++i) {
-      const verseKey = verseKeys[i];
-      const verseLyrics = lyricBlocks[i];
+    // if the song has a presentation order defined, use it;
+    // otherwise, present in written order
+    if (typeof presentationOrder !== "undefined") {
```

Comment:
should this be `if (presentationOrder != null)` ?

Reply:
Ahh yeah that's probably safer